gap/LazyCategory.gi

```diff
@@ -754,18 +754,20 @@ InstallGlobalFunction( CAP_INTERNAL_COMPACT_NAME_OF_CATEGORICAL_OPERATION,
     posWith := PositionSublist( name, "WithGiven" );
     posFrom := PositionSublist( name, "From" );
     posInto := PositionSublist( name, "Into" );
-    posOf := PositionsSublist( name, "Of" );
+    # find last occurrence of "Of"
+    for i in [ Length( name ) - 2 .. 0 ] do
+        posOf := PositionSublist( name, "Of", i );
+        if posOf <> fail then
+            break;
+        fi;
+    od;
     posHom := PositionSublist( name, "HomomorphismStructure" );
     
-    if not IsEmpty( posOf ) then
-        posOf := posOf[Length( posOf )];
-    fi;
-    
     cname := name;
     
     if IsInt( posWith ) then
         if posWith > l / 2 then
-            if IsInt( posOf ) then
+            if posOf <> fail then
```

Comment:
Did you need to change this line?

Reply:
No, but at least for me this way the code is easier to understand: If I read `IsInt( posOf )` I ask myself "What other type could posOf possibly have?". If I read "posOf <> fail" I immediately know that we exclude the corner case (resp. handle it in the else case).